operator/metrics/metrics.go

```diff
@@ -32,7 +33,8 @@ type params struct {
 
 	Metrics []metric.WithMetadata `group:"hive-metrics"`
 
-	Registry *metrics.Registry
+	Registry         *metrics.Registry
+	TlsConfigPromise tlsConfigPromise
```

Comment:
```suggestion
	PrometheusTlsConfigPromise prometheusTLSConfigPromise
```
Let's specify this is for prometheus.